lib/trento/support/struct_helper.ex

```diff
@@ -33,4 +33,41 @@ defmodule Trento.Support.StructHelper do
   def to_map(value) when is_nil(value), do: value
   def to_map(value) when is_atom(value), do: Atom.to_string(value)
   def to_map(value), do: value
+
+  @doc """
+  Converts the string keys of a map to existing atoms.
+  If the key does not exist as atom it continues being a string
+  """
+  @spec to_atomized_map(map | [map] | struct | [struct]) :: map | [map]
+  def to_atomized_map(%NaiveDateTime{} = value), do: value
+  def to_atomized_map(%DateTime{} = value), do: value
+  def to_atomized_map(%Date{} = value), do: value
+
+  def to_atomized_map(struct) when is_struct(struct) do
+    struct
+    |> Map.from_struct()
+    |> to_atomized_map()
+  end
+
+  def to_atomized_map(map) when is_map(map) do
+    map
+    |> Enum.reject(fn
+      {:__meta__, _} -> true
+      {_, %Ecto.Association.NotLoaded{}} -> true
+      _ -> false
+    end)
+    |> Map.new(fn {k, v} -> {atomize_key(k), to_atomized_map(v)} end)
+  end
+
+  def to_atomized_map(list) when is_list(list) do
+    Enum.map(list, &to_atomized_map/1)
+  end
+
+  def to_atomized_map(value), do: value
+
+  defp atomize_key(key) do
+    String.to_existing_atom(key)
+  rescue
```

Comment:
This is the unique way I found of knowing if an atom exists or not.
Unfortunately I didn't find a way to remove the `try/rescue` option